staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -1207,7 +1207,7 @@ func validateMapListKeysMapSet(schema *apiextensions.JSONSchemaProps, fldPath *f
 				continue
 			}
 
-			if isRequired[k] == false && obj.Default == nil {
+			if isRequired[k] == false && (obj.Default == nil || obj.Default.Object == nil) {
```

Comment:
any other places we're checking nil for any JSON fields (`example`, `enum`, `default`) that will need an addition like this to check the nested object for nil?